File: dotnet_distance_generator/trunk/src/csharp/tests/TopCoder/Web/Distance/AccuracyTests/OverlapDistanceCalculatorTests.cs
    using System;
    using System.Collections.Generic;
    using System.Text;
    using NUnit.Framework;
    using TopCoder.Web.Distance.Data;
    using TopCoder.Web.Distance.DistanceCalculators;


namespace TopCoder.Web.Distance.AccuracyTests
{
    [TestFixture, CoverageExclude]
    public class OverlapDistanceCalculatorTests
    {
        /// <summary>
        /// Tests for both defined distances
        /// </summary>
        [Test]
        public void TestBothDefined()
        {
            IDistanceCalculator calculator = new OverlapDistanceCalculator();
            IList<float> list = calculator.CalculateDistance(AccuracyTestHelper
                    .GetTestDataAccess().GetMember(AccuracyTestHelper.MEMBER_IDS[0]),
                    AccuracyTestHelper.GetTestDataAccess().GetRelatedMembers(
                            AccuracyTestHelper.MEMBER_IDS[0], AccuracyTestHelper.GetAllEnum()), AccuracyTestHelper
                            .GetAlgoritmEnum());

            Assert.AreEqual((float)(1.0 - (33.0 - 19.0) / 49.0), list[0]);
        }

        /// <summary>
        /// Tests for both defined distances
        /// </summary>
        [Test]
        public void TestBothDefined2()
        {
            IDistanceCalculator calculator = new OverlapDistanceCalculator();
            IList<float> list = calculator.CalculateDistance(AccuracyTestHelper
                    .GetTestDataAccess().GetMember(AccuracyTestHelper.MEMBER_IDS[0]),
                    AccuracyTestHelper.GetTestDataAccess().GetRelatedMembers(
                            AccuracyTestHelper.MEMBER_IDS[0], AccuracyTestHelper.GetAllEnum()), AccuracyTestHelper
                            .GetAlgoritmEnum());

            Assert.AreEqual((float)(1.0 - (49.0 - 19.0) / 49.0), list[1]);
        }

        /// <summary>
        /// Tests for zero max distance
        /// </summary>
        [Test]
        public void TestZeroMaxDistance()
        {
            IDistanceCalculator calculator = new OverlapDistanceCalculator();
            IList<float> list = calculator.CalculateDistance(AccuracyTestHelper
                    .GetTestDataAccess().GetMember(156859),
                    AccuracyTestHelper.GetTestDataAccess().GetRelatedMembers(
                            156859, AccuracyTestHelper.GetAllEnum()), AccuracyTestHelper
                            .GetAlgoritmEnum());

            Assert.AreEqual(1.0f, list[1]);
        }
    }
}

File: java_distance_generator/trunk/src/java/tests/com/topcoder/web/tc/distance/accuracytests/OverlapDistanceCalculatorTests.java
package com.topcoder.web.tc.distance.accuracytests;

import java.util.List;

import junit.framework.TestCase;

import com.topcoder.web.tc.distance.DistanceCalculator;
import com.topcoder.web.tc.distance.calculators.MatchOverlapDistanceCalculator;
import com.topcoder.web.tc.distance.data.Member;

/**
 * Accuracy test cases for OverlapDistanceCalculator.
 *
 * @author aksonov
 *
 */
public class OverlapDistanceCalculatorTests extends TestCase {

    /**
     * Tests for both defined distances.
     */
    public void testBothDefined() {
        DistanceCalculator calculator = new MatchOverlapDistanceCalculator();
        List<Double> list = calculator.getDistance(
            TestHelper.getTestDataAccess().getMember(TestHelper.MEMBER_IDS[0]), TestHelper.getTestDataAccess()
                .getRelatedMembers(TestHelper.MEMBER_IDS[0], null), TestHelper.getAlgoritmEnum());

        // min overlap 19
        // max overlap 49
        // overlap is 33
        // result is 1 - (33 - 19)/49
        assertEquals(1.0 - (33.0 - 19.0) / 49.0, list.get(0), 1e-6);
    }

    /**
     * Tests for both defined distances.
     */
    public void testBothDefined2() {
        DistanceCalculator calculator = new MatchOverlapDistanceCalculator();
        List<Double> list = calculator.getDistance(
            TestHelper.getTestDataAccess().getMember(TestHelper.MEMBER_IDS[0]), TestHelper.getTestDataAccess()
                .getRelatedMembers(TestHelper.MEMBER_IDS[0], null), TestHelper.getAlgoritmEnum());

        // min overlap 19
        // max overlap 49
        // overlap is 33
        // result is 1 - (49 - 19)/49
        assertEquals(1.0 - (49.0 - 19.0) / 49.0, list.get(1), 1e-6);
    }

    /**
     * Tests for zero max distance.
     */
    public void testZeroMaxDistance() {
        DistanceCalculator calculator = new MatchOverlapDistanceCalculator();
        List<Member> members = TestHelper.getTestDataAccess().getRelatedMembers(156859,
            TestHelper.getAlgoritmDevEnum());

        List<Double> list = calculator.getDistance(members.get(0), members, TestHelper.getAlgoritmEnum());

        assertEquals(1.0, list.get(1));
    }

}
